core/src/main/java/org/elasticsearch/action/index/TransportIndexAction.java

```diff
@@ -88,6 +90,9 @@ protected void doExecute(final IndexRequest request, final ActionListener<IndexR
         // if we don't have a master, we don't have metadata, that's fine, let it find a master using create index API
         ClusterState state = clusterService.state();
         if (autoCreateIndex.shouldAutoCreate(request.index(), state)) {
+            if (!settings.getAsBoolean(MapperService.DYNAMIC_MAPPING_ENABLED_SETTING, MapperService.DYNAMIC_MAPPING_ENABLED_DEFAULT)) {
+                throw new MapperParsingException("trying to auto create mapping, but dynamic mapping is disabled");
```

Comment:
TypeMissingException ? It's a not a parsing exception per say. Can you add the index name and the type in the message.